### ISSUE TITLE
When activating virtual sessions and pressing the controller button, the logged-out host PC screen turns on.

### ISSUE THREAD

**Issue by Protomothis** [DuoStream/Duo#117]
It may seem obvious, but when making controller inputs on a Steam Deck connected to a virtual session, the screen of the logged-out host PC turns on.
The controller connected to the virtual session does not affect all sessions. Actual inputs are not directly transmitted. However, is it possible to prevent the monitor screen from staying on when the virtual session is activated?
I'm curious if there's a way to connect to the virtual session while putting the host PC monitor to sleep.

**Comment by appledoc1** [comment]
You can always turn off the host PC monitor when you don't need it, it wont affect Duo. 
But I've experienced the same thing where the Duo session will wake the host monitor from standby mode or from a screensaver. It would be great if it could be avoided.

**Comment by Protomothis** [comment]
@appledoc1 yeah If possible, I'd prefer if the host PC monitor doesn't respond, rather than manually turning off the monitor power. It's frustrating when the monitor flickers every time there's input...

**Comment by samkitty** [comment]
Might be a silly solution but there are apps that display a fully black screen when you press a hotkey like [this one](https://apps.microsoft.com/detail/9nrtgl0jzd01?hl=en-US&gl=US). You can make an autohotkey script that triggers it on your host PC whenever Duo is in use and disables it whenever you're done. Alternatively you can use [MultiMonitorTool](https://www.nirsoft.net/utils/multi_monitor_tool.html), which allows you to turn your monitor on/off and change resolution refresh rate etc using command line (read the description for details). You can have it setup so the that only the virtual monitor (which Duo uses) is turned on while using Duo.

**Comment by Protomothis** [comment]
@samkitty Thanks for helping me, sam. I'm hoping DuoStream will help me do it naturally. Am I being too lazy? 😅

**Comment by samkitty** [comment]
You're welcome! And no I think it's a valid request.

**Comment by Black-Seraph** [comment]
Microsoft doesn't provide virtual input drivers for gamepads, only for mouse and keyboard, which means we are forced to go through third party drivers like ViGEMBus to spawn gamepads. 

And because of Microsoft's strict driver signature guidelines we are unable to sign our own drivers to potentially replace ViGEMBus...

This leaves our hands tied with no real way to fix this issue until Microsoft provides a suitable remoting-compatible replacement driver. 

tl;dr: Issue acknowledged, but no way to fix it, thus closed. :(

**Comment by dalicode** [comment]
> Might be a silly solution but there are apps that display a fully black screen when you press a hotkey like [this one](https://apps.microsoft.com/detail/9nrtgl0jzd01?hl=en-US&gl=US). You can make an autohotkey script that triggers it on your host PC whenever Duo is in use and disables it whenever you're done. Alternatively you can use [MultiMonitorTool](https://www.nirsoft.net/utils/multi_monitor_tool.html), which allows you to turn your monitor on/off and change resolution refresh rate etc using command line (read the description for details). You can have it setup so the that only the virtual monitor (which Duo uses) is turned on while using Duo.

How are you running a script on host PC when detecting game launch?

**Comment by samkitty** [comment]
> @samkitty How are you running a script on host PC when detecting game launch?

Not on game launch, I use these webhook calls as a simple http request in Tasker to start/stop my Duo instance whenever I open/close Moonlight app.
`http://127.0.0.1:38299/instances/<instance name>/start` and `http://127.0.0.1:38299/instances/<instance name>/stop` then create a task using task scheduler triggered on Log On of the Duo user (or a .bat file in startup folder), that way my scripts run as soon as Duo starts.

Alternatively, you can paste the following line into the Launch Options of a game on Steam to run a script file alongside your games (in this case I'm launching notepad.exe):

`"C:\Windows\System32\notepad.exe" %COMMAND%`
